src/test/scala/cognite/spark/v1/SequenceRowsRelationTest.scala

```diff
@@ -0,0 +1,243 @@
+package cognite.spark.v1
+
+import cats.data.NonEmptyList
+import com.cognite.sdk.scala.common.CdpApiException
+import com.cognite.sdk.scala.v1.{Sequence, SequenceColumn, SequenceColumnCreate}
+import io.scalaland.chimney.dsl._
+import org.apache.spark.sql.{DataFrame, Row}
+import org.apache.spark.sql.functions.col
+import org.apache.spark.SparkException
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.util.control.NonFatal
+
+class SequenceRowsRelationTest extends FlatSpec with Matchers with SparkTest {
+  import spark.implicits._
+
+  private val sequencesSourceDf = spark.read
+    .format("cognite.spark.v1")
+    .option("apiKey", writeApiKey)
+    .option("type", "sequences")
+    .load()
+  sequencesSourceDf.createOrReplaceTempView("sequences")
+
+  val sequenceA = SequenceUpdateSchema(
+    externalId = Some("a"),
+    name = Some("Rows test sequence"),
+    columns = Some(Seq(
+      SequenceColumnCreate(
+        externalId = "num1",
+        valueType = "LONG"
+      ),
+      SequenceColumnCreate(
+        externalId = "str1",
+        valueType = "STRING"
+      ),
+      SequenceColumnCreate(
+        externalId = "num2",
+        valueType = "DOUBLE"
+      )
+    ))
+  )
+  val sequenceB = SequenceUpdateSchema(
+    externalId = Some("b"),
+    name = Some("Rows test many sequence"),
+    columns = Some(
+      Seq(
+        SequenceColumnCreate(
+          externalId = "num1",
+          valueType = "LONG"
+        )
+      ))
+  )
+
+  it should "create and read rows" in withSequences(Seq(sequenceA)) { case Seq(sequenceId) =>
+
+    spark.sparkContext
+      .parallelize(1 to 100)
+      .toDF()
+      .createOrReplaceTempView("numbers")
+
+    insertRows(
+      sequenceId,
+      spark
+        .sql("select value as rowNumber, 'abc' as str1, 1.1 as num2, value * 6 as num1 from numbers"))
+
+    val allColumns = retryWhile[Array[Row]](
```

Comment:
Sequences are *not* eventually consistent, so the data should be available immediately, but I guess the retry doesn't hurt.

Reply:
Are you sure? I have just tried, and it seems that it just does not return any rows right after the insert :/ I'll just keep the retries here, but would be nice to remove them.

Reply:
Ah, it's like data points... the storage is not eventually consistent, but there's a pubsub queue in the way, so for the user it is indeed eventually consistent. :/